packages/inferno/src/core/component.ts

```diff
@@ -130,7 +130,7 @@ export interface ComponentLifecycle<P, S> {
 export interface Component<P = {}, S = {}> extends ComponentLifecycle<P, S> {}
 export class Component<P, S> {
   // Public
-  public static defaultProps: {} | null = null;
+  public static defaultProps?: Partial<any>;
```

Comment:
Isn't Partial<P> technically more correct? Partial props? This is also how React implemented it.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/v15/index.d.ts#L350

Reply:
`defaultProps` is a `static` member here, that would cause another error

```
[ts] Static members cannot reference class type parameters.
```